src/scheduler/limits.cpp

```diff
@@ -1768,17 +1717,16 @@ check_queue_max_res(server_info *si, queue_info *qi, resource_resv *rr,
 	schd_resource	*res;
 	resource_count	*used_res;
 	counts		*c;
-	counts		*cts = NULL;
 
 	if ((qi == NULL) || (rr == NULL))
 		return (SCHD_ERROR);
 
-	if ((qc == NULL) || (qc->all == NULL))
+	if ((qc == NULL) || (qc->all.size() == 0))
```

Comment:
no need to check for size().  It'll just fail to find something and return 0 further down.  The original check against NULL was to guard against crashing.